bootstrap.sh

```diff
@@ -120,165 +307,111 @@ echo "Enter your domain name"
 echo "The domain name should already resolve to the IP address of your server"
 echo "Make sure that 'wg' and 'auth' subdomains also point to that IP (not necessary with DuckDNS)"
 echo
-read -p "Domain name: " root_host
-until [[ "$root_host" =~ ^[a-z0-9\.\-]*$ ]]; do
-  echo "Invalid domain name"
-  read -p "Domain name: " root_host
+root_host=
+until [[ ${root_host} =~ ^[a-z0-9\.-]*$ && -n ${root_host} ]]; do
+	[[ -n ${root_host} ]] && echo "Invalid domain name"
+	read -r -p "Domain name: " root_host
 done
 
-public_ip=$(curl -s ipinfo.io/ip)
-domain_ip=$(dig +short @1.1.1.1 ${root_host})
-
-until [[ $domain_ip =~ $public_ip ]]; do
-  echo
-  echo "The domain $root_host does not resolve to the public IP of this server ($public_ip)"
-  echo
-  root_host_prev=$root_host
-  read -p "Domain name [$root_host_prev]: " root_host
-  if [ -z ${root_host} ]; then
-    root_host=$root_host_prev
-  fi
-  public_ip=$(curl -s ipinfo.io/ip)
-  domain_ip=$(dig +short @1.1.1.1 ${root_host})
-  echo
-done
 
 echo
-echo "Running certbot in dry-run mode to test the validity of the domain..."
-$SUDO certbot certonly --non-interactive --break-my-certs --force-renewal --agree-tos --email root@localhost.com --standalone --staging -d $root_host -d wg.$root_host -d auth.$root_host || exit
-echo "OK"
-
-echo "root_host: \"${root_host}\"" >> $HOME/ansible-easy-vpn/custom.yml
-
-
-if [[ ! $aws =~ true ]]; then
-  echo
-  echo "Would you like to use an existing SSH key?"
-  echo "Press 'n' if you want to generate a new SSH key pair"
-  echo
-  read -p "Use existing SSH key? [y/N]: " new_ssh_key_pair
-  until [[ "$new_ssh_key_pair" =~ ^[yYnN]*$ ]]; do
-          echo "$new_ssh_key_pair: invalid selection."
-          read -p "[y/N]: " new_ssh_key_pair
-  done
-  echo "enable_ssh_keygen: true" >> $HOME/ansible-easy-vpn/custom.yml
-
-  if [[ "$new_ssh_key_pair" =~ ^[yY]$ ]]; then
-    echo
-    read -p "Please enter your SSH public key: " ssh_key_pair
-
-    echo "ssh_public_key: \"${ssh_key_pair}\"" >> $HOME/ansible-easy-vpn/custom.yml
-  fi
-else
-  echo
-  read -p "Are you running this script on an AWS EC2 instance? [y/N]: " aws_ec2
-  until [[ "$aws_ec2" =~ ^[yYnN]*$ ]]; do
-          echo "$aws_ec2: invalid selection."
-          read -p "[y/N]: " aws_ec2
-  done
-  if [[ "$aws_ec2" =~ ^[yY]$ ]]; then
-    export AWS_EC2=true
-  echo
-  echo "Please use the SSH keys that you specified in the AWS Management Console to log in to the server."
-  echo "Also, make sure that your Security Group allows inbound connections on 51820/udp, 80/tcp and 443/tcp."
-  echo
-  fi
-fi
+[[ -f /etc/resolv.conf ]] && {
+	echo "Entries in /etc/resolv.conf for nameserver(s)"
+	grep ^nameserver /etc/resolv.conf
+	echo
+}
+read -r -p "Enter the IP address of the Nameserver to use for DNS queires [1.1.1.1]: " DNS_HOST
```

Comment:
Not sure if this is the way to go. I will probably implement a DNS choice later, but asking the user for DNS for a `dig` seems superfluous

Reply:
> Not sure if this is the way to go. I will probably implement a DNS choice later, but asking the user for DNS for a `dig` seems superfluous

But "forcing" 1.1.1.1 choice ..... I get it, choice isn't always the answer, but the script should probably not try to use any specific service and leave it up to the user's own normal config.

Reply:
That would be my first choice as well – but it turns out that using system's resolvers results in the `dig` results being 'cached' (maybe something to do with systemd-resolved?), so using an authoritative server helps: https://serverfault.com/questions/372066/force-dig-to-resolve-without-using-cache